ui/desktop/app/routes/scopes/scope/projects/targets/target.js

```diff
@@ -23,6 +23,56 @@ export default class ScopesScopeProjectsTargetsTargetRoute extends Route {
 
   // =methods
 
+  /**
+   * Given an array of hostSources returns an array of host sets
+   * @param {array} hostSources
+   * @returns {[HostSetModel]} hostSets
+   */
+  async getTargetHostSets(hostSources) {
+    let hostSets = [];
+
+    for (const hostSource of hostSources) {
+      const { host_source_id } = hostSource;
+      try {
+        const hostSet = await this.store.findRecord('host-set', host_source_id);
+        hostSets.push(hostSet);
+      } catch (e) {
+        // IF error is a lack of permissions, stop fetching host-sets.
+        if (e.errors[0].isForbidden) {
+          break;
+        }
+      }
+    }
+    return hostSets;
+  }
+
+  /**
+   * Given an array of hostSets returns an array of all the hosts within the
+   * host-set
+   * @param {array} hostSets
+   * @returns {[HostModel]} hosts
+   */
+  async getHostsFromHostSets(hostSets) {
+    const hosts = [];
+
+    if (hostSets.length > 1) {
```

Comment:
Why is this length check needed?

Reply:
@ZedLi 
The purpose of this was to check hostSets has actually content. But you are right, despite I think the check should stay, the way of checking is incorrect. If there is just 1 hostSet this will not enter to fetch it, so it's incorrect.
I will change the check you just `if (hostSets)`. If hostSets is empty, we will return an empty array (so will not fail) but if hostSets has any value we will try to fetch them.